site/src/pages/RoadmapPage/Planner.tsx

```diff
@@ -112,14 +131,42 @@ const Planner: FC = () => {
 
       // check current roadmap against last-saved roadmap in local storage
       // if they are different, mark changes as unsaved to enable alert on page leave
-      dispatch(setUnsavedChanges(localStorage.getItem('roadmap') !== roadmapStr));
+      const localRoadmap = JSON.parse(localStorage.getItem('roadmap') ?? emptyRoadmap);
+      delete localRoadmap.timestamp;
+      dispatch(setUnsavedChanges(JSON.stringify(localRoadmap) !== roadmapStr));
     }
   }, [cookies, currentPlanData, dispatch, isFirstRenderer, roadmapStr, transfers]);
 
   const { unitCount, courseCount } = calculatePlannerOverviewStats();
 
   return (
     <div className="planner">
+      <Modal
+        show={showSyncModal}
+        onHide={() => {
+          setShowSyncModal(false);
+        }}
+        className="ppc-modal"
+        centered
+      >
+        <Modal.Header closeButton>
+          <h2>Cloud roadmap out-of-date</h2>
+        </Modal.Header>
+        <Modal.Body>
+          <p>
+            Your roadmap saved locally is newer than the one saved to the cloud. Would you like to use the local one
+            instead?
+          </p>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button variant="danger" onClick={handleLoadLocal}>
+            Yes, Use Local
+          </Button>
+          <Button variant="secondary" onClick={() => setShowSyncModal(false)}>
+            No, Use Cloud
```

Comment:
Consider which one you want to be the primary action. Both are potentially destructive, but in the case that this popup occurs, do you think it's more likely that the user "just got signed out and wants to use their local one"? (That's what I imagine, but not sure if you agree)

If you agree, I would make "Yes" -> "This Device" and change the color to the primary, and change "No, Use Cloud" -> "My Account"

Reply:
Yes, the scenario I wrote this issue for was when making changes to my roadmap locally without realizing I'm signed out and having no way to load them onto my account. The only other scenario where I imagine this modal appearing is if multiple people are logging in/out of the same device which I don't see happening often.